python/lsst/ts/standardscripts/auxtel/detector_characterization/get_std_flat_dataset.py

```diff
@@ -0,0 +1,279 @@
+
+        if self.grating is not None:
+            await self.checkpoint(f"Configuring ATSpec grating; {self.grating}")
+            self.atspec.cmd_changeDisperser.set(disperser=self.grating)
+            await self.atspec.cmd_changeDisperser.start(timeout=self.cmd_timeout)
+
+        if self.linear_stage is not None:
+            await self.checkpoint(f"Configuring ATSpec linear stage; {self.linear_stage}")
+            self.atspec.cmd_moveLinearStage.set(distanceFromHome=self.linear_stage)
+            await self.atspec.cmd_moveLinearStage.start()
+
+        await self.take_dark_sequence()
+
+        await self.take_bias_sequence()
+
+        await self.take_flat_sequence()
+
+        await self.take_bias_sequence()
+
+        await self.checkpoint("done")
+
+    async def take_dark_sequence(self):
+        """A coroutine to take the intended sequence of darks.
+        """
+        self.log.debug("Taking dark sequence...")
+
+        for i in range(self.n_dark):
+            await self.take_image(exp_time=self.t_dark,
+                                  shutter=False,
+                                  image_seq_name=f"dark_{i+1:04d}_{self.n_dark:04d}")
+
+    async def take_bias_sequence(self):
+        """A coroutine to take the intended sequence of bias.
+        """
+        self.log.debug("Taking bias sequence...")
+
+        for i in range(self.n_bias):
+            await self.take_image(exp_time=0.,
+                                  shutter=False,
+                                  image_seq_name=f"bias_{i+1:04d}_{self.n_bias:04d}")
+
+    async def take_flat_sequence(self):
+        """A coroutine to take the intended sequence of flats.
+        """
+        self.log.debug("Taking flat sequence...")
+
+        for i in range(len(self.flat_dn_range)):
+            for j in range(self.n_flat):
+                await self.take_image(exp_time=self.flat_base_exptime * self.flat_dn_range[i],
+                                      shutter=True,
+                                      image_seq_name=f"flat_{i+1:04d}_"
+                                                     f"{len(self.flat_dn_range):04d}_"
+                                                     f"{j+1:04d}_"
+                                                     f"{self.n_flat:04d}")
+
+    async def take_image(self, exp_time, shutter, image_seq_name):
+        """Coroutine to take images.
+
+        Parameters
+        ----------
+        exp_time : `float`
+            The exposure time for the image, in seconds.
+        shutter : `bool`
+            Should activate the shutter? (False for bias and dark)
+        image_seq_name : `str`
+            A string to identify the image.
+
+        Returns
+        -------
+
+        """
+        self.atcam.cmd_takeImages.set(numImages=1,
+                                      expTime=exp_time,
+                                      shutter=shutter,
+                                      imageSequenceName=str(image_seq_name))
```

Comment:
I am curious why you choose to have take_image take just one image, instead of all the images for a sequence with a single call to takeImages (by setting numImages > 1). Of course what you have will work just fine. But other readers may have the same question, so please consider a code comment explaining.

Reply:
This is due to an issue with the current version of ATCamera which is not set to take images with `numImages>1`... I added a comment to the code...